app/services/docker/containerService.js

```diff
@@ -149,6 +149,23 @@ angular.module('portainer.services')
 
     return deferred.promise;
   };
+  
+  service.inspect = function(id) {
```

Comment:
I don't think there is a need to use `$q.defer()` here. You could simply use:

```javascript
service.inspect = function(id) {
  return Container.inspect({id: id}).$promise;
};
```

Reply:
Agree, I just found that it's done the same way for other functions in this service, may I refactor them as well?

Reply:
Sure, but I believe there should not be many functions looking the same. Usually you'll find some process in the `function success()`.